src/webpack.config.js

```diff
@@ -0,0 +1,100 @@
+import ViewEngine from './viewEngine';
+import {BundleAnalyzerPlugin} from 'webpack-bundle-analyzer';
+import path from 'path';
+import webpack from 'webpack';
+import nconf from 'nconf';
+
+nconf.argv().env().file({file: path.join(__dirname, '../config.json')}).overrides({
+  APP_ENV: 'browser',
+});
+
+const GROUPS = {};
```

Comment:
Each file in `admin/frontend/views` and `common/frontend/views` is a target that we want to compile -- this is the script that gets linked in every respective page. A home page script for the home page, an article script for an article page, etc.

However, lots of these have shared code so we want to have some scripts that bundle together the stuff that they share. So we define groups which are these shared bundles. Currently there's one group for common code and one group for admin code. These get created in the webpack config on line 94 of this file. Then, the `Renderer` knows which group a target belongs to so that when it generates the page it can add the correct script for the shared code.